lib/elasticsearch.ts

```diff
@@ -0,0 +1,38 @@
+var config  = require('../config'); 
+var request = require('request');
+
+
+// Helper function to make ES queries. Pass it a path to query and it returns
+// a promise with the decoded result
+var _get_es = function(path) {
+    // wrap the async request in a promise and return it
+    return new Promise(function(resolve, reject) {
+        var url = config.ELASTICSEARCH_URL + path;
+        var auth = {
+            'user': config.ELASTICSEARCH_USER,
+            'pass': config.ELASTICSEARCH_PASSWORD,
+            'sendImmediately': false
+        };
+        request.get({'url': url, 'auth': auth}, function(error, response, body) {
+            if (!error && response.statusCode == 200) {
+                resolve(JSON.parse(body));
+            } else {
+                reject(Error(response.statusText));
+            }
+        });
+    });
+}
+
+// Returns a promise with a list of all the indexes
+// TODO: extend this to also include shards and other interesting data
+export function get_indexes() {
+    return _get_es('_stats?level=shards').then(function(data) {
+        // parse the index names into a list
+        var indexes = [];
+        Object.keys(data['indices']).forEach(function(key) {
+            indexes.push(key);
+        });
+        return indexes;
+    });
+}
+
```

Comment:
would be nice to get a pattern for mocking/testing in place with this initial PR